consensus/core/src/linearizer.rs

```diff
@@ -84,6 +96,12 @@ impl Linearizer {
 
         drop(dag_state);
 
+        // The above code should have not yielded any blocks that are <= gc_round, but just to make sure that we'll never
+        // commit anything that should be garbage collected we attempt to prune here as well.
+        if gc_enabled {
```

Comment:
Is this really necessary? Maybe have this as an assertion instead?

Reply:
Not necessary and I'll turn this into an assertion indeed.